### PR TITLE
Miscellaneous bug fixes

### LINKED ISSUE #39: Invalidate non-numerical errors
The `symerror` and `plus`/`minus` components of `asymerror` can be either a `number` or a `string`.  The latter is needed to support percentage uncertainties specified by a number followed by a `%` symbol, and also for one-sided uncertainties where either `plus` or `minus` is given as an empty string `''`.  Other cases should be invalidated, that is, if `symerror`, `plus` or `minus` cannot be converted to a float after possibly removing a terminating `%` symbol, and one of `plus` or `minus` (but not both) is not an empty string `''`, then a validation error should be returned.

[Here](https://www.hepdata.net/record/ins793126?version=1&table=Figure%2017a%20-%20TOF%20data) is an example table which has entries like `symerror: 2.300e-003f` in the YAML file and so should have been invalidated at the upload stage.  This example was raised in a [comment](https://gitlab.com/hepcedar/rivet/-/issues/246#note_720234019) on a Rivet issue.

### LINKED ISSUE #40: Only make additional Python checks if no schema errors
I've noticed a few recent error emails from the submission system with `An unexpected error occurred: float() argument must be a string or a number, not 'NoneType'` and an exception raised in Sentry ([example](https://hepdata-sentry.web.cern.ch/sentry/hepdata-prod/issues/29742/?environment=production&referrer=alert_email)) with `Final attempt of process_saved_file for recid 1636571704 failed. Resetting to previous status.`  This happens because null values are given for some fields, e.g.
```yaml
dependent_variables:
- header: {name: '$d^{2}\sigma/d\it{p}_{T}d\eta(mb \it{c}/GeV)$'}
  qualifiers:
  - {name: '$\it{p}_{T,jet}^{ch} (GeV/c)$', value: 'Jetptg R=0.2'}
  values:
  - value: 
    errors:
    - {symerror: , label: '$p_{T}$ statistical'}
    - {symerror: , label: '$p_{T}$ systematic'}
  - value: 
    errors:
    - {symerror: , label: '$p_{T}$ statistical'}
    - {symerror: , label: '$p_{T}$ systematic'}
```
Here, the `value` and `symerror` will show as `None` in Python, giving an uncaught exception `TypeError: float() argument must be a string or a number, not 'NoneType'` from [`error = float(error)`](https://github.com/HEPData/hepdata-validator/blob/9c38835252c4b615b9100a09cf5e32902c80b000/hepdata_validator/data_file_validator.py#L260).  Probably it would be easier just to make the additional Python checks such as `check_for_zero_uncertainty` if there are no validation errors from the JSON schema (i.e. revert a change made in https://github.com/HEPData/hepdata-validator/commit/dd5d463de77898579f1f108d5dbe711f694c5d88).  Otherwise, the Python code making the additional checks cannot make any assumptions about the validity of the fields.

Could you please address this issue, @alisonrclarke, together with #39?

### LINKED ISSUE #42: Fix check of cmenergies given as a range
The current code uses a regex:

https://github.com/HEPData/hepdata-validator/blob/9c38835252c4b615b9100a09cf5e32902c80b000/hepdata_validator/submission_file_validator.py#L179

which only allows one digit after a decimal point.  This invalidates legitimate entries like `0.03-3.0`.  The regex should be modified to allow any number of digits after a decimal point.  Could you please address this in PR #41, @alisonrclarke?

### PR DESCRIPTION
**Pull request by alisonrclarke**
* Fixes #39 
 * Fixes #40
 * Fixes #42

### CODE REVIEW

File: hepdata_validator/data_file_validator.py [hunk 1]
```diff
@@ -163,58 +174,72 @@ def check_independent_variable_values(self, file_path, data_item):
...
+                                instance=data['dependent_variables']
+                            )
+                            if error_plus == '' and error_minus == '':
                                 error = ValidationError(
-                                    "Uncertainties should not all be zero",
-                                     path=['dependent_variables', 'values', i, 'errors'],
-                                     instance=data['dependent_variables']
+                                    "asymerror plus and minus cannot both be empty",
+                                    path=['dependent_variables', 'values', i, 'errors', 'asymerror'],
+                                    instance=data['dependent_variables']
```
Comment (GraemeWatt) [inline hepdata_validator/data_file_validator.py:226]: Can you please also invalidate `symerror: ''` (an empty string)?  That case is equivalent to an `asymerror` with `plus` and `minus` both set to an empty string.

File: hepdata_validator/data_file_validator.py [hunk 2]
```diff
@@ -223,17 +248,37 @@ def check_length_values(self, file_path, data):
...
+        :return: error as a float if possible, otherwise the original string
+        """
+        if isinstance(error, str):
+            error = error.replace('%', '')  # strip percentage symbol
+        try:
+            error = float(error)
+        except ValueError:
+            if error != '':  # empty string is allowed in some circumstances
+                validation_error = ValidationError(
+                    f"Invalid error value {error}: value must be a float (with or without %) or empty string",
```
Comment (GraemeWatt) [inline hepdata_validator/data_file_validator.py:275]: Maybe change the message after the colon to "value must be a number (possibly ending in %)", since the empty-string case is not allowed in general, only for the `asymerror` case for one of `plus` or `minus`.

File: testsuite/test_data/invalid_file.yaml [hunk 3]
```diff
@@ -16,5 +16,6 @@ dependent_variables:
       - value: 5.7
         errors:
           - {symerror: 0.4, label: 0.443}
-          - {asymerror: {plus: 0.42, minus: 0.31}, label: sys}
-          - {symerror: 0.4, label: "sys,lumi"}
\ No newline at end of file
+          - {asymerror: {plus: 0.42%, minus: 0.31%}, label: sys}
+          - {symerror: 2.300e-003f, label: "sys,lumi"}
+          - {asymerror: {plus: '', minus: ''}, label: stat}
```
Comment (GraemeWatt) [inline testsuite/test_data/invalid_file.yaml:21]: Add a test for `symerror: ''` (an empty string).

**Review by GraemeWatt — APPROVED** [review]
Thanks, looks good!  Just one main request to invalidate the case of `symerror: ''` (an empty string).

**Review by GraemeWatt — APPROVED** [review]
Thanks, looks good!
